web/app/Mail/MJMLMailable.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Wikijump\Mail;
+
+use Illuminate\Mail\Mailable;
+use Wikijump\Services\MJML\MJML;
+
+/**
+ * Mailable that supports using MJML.
+ *
+ * Use the `mjml` method to set the MJML content.
+ */
+class MJMLMailable extends Mailable
+{
+    /** MJML template path. */
+    protected string $mjml;
+
+    /**
+     * Sets the MJML template to use. This template has the highest priority.
+     * Templates will be rendered by Blade first, and then by MJML.
+     *
+     * @param string $mjml Path to the MJML template, in Blade template syntax..
+     * @param array $data Data to pass to the template.
+     */
+    public function mjml(string $mjml, array $data = []): self
```

Comment:
`: self` works?

Reply:
Apparently? I was trying to figure out how to type `$this`, but that can't be in that "returns" slot in the function signature, but `self` can go in there. Although, now that I think about it, that breaks polymorphism... ugh, I'll change that.

Reply:
Honestly I think putting the class name there again is more readable anyways.

Reply:
Yeah, but that also breaks polymorphism. You have to use a doc comment with `@return $this` for it to work correctly. It feels kind of weird that `self` is valid but `this` isn't... PHP things I guess

Reply:
Lol alright makes sense